server/src/test/java/org/elasticsearch/index/mapper/NumberFieldTypeTests.java

```diff
@@ -68,6 +68,17 @@ protected MappedFieldType createDefaultFieldType() {
         return new NumberFieldMapper.NumberFieldType(type);
     }
 
+    public void testEqualityWithDifferentNumberTypes() {
+        NumberType type = randomFrom(NumberType.values());
+        NumberFieldType fieldType = new NumberFieldType(type);
+
+        NumberType otherType = randomValueOtherThan(type,
+            () -> randomFrom(NumberType.values()));
+        NumberFieldType otherFieldType = new NumberFieldType(otherType);
+
+        assertNotEquals(fieldType, otherFieldType);
```

Comment:
If you wanted, this could be tested with `EqualsHashCodeTestUtils.checkEqualsAndHashCode` for testing the methods and mutation

Reply:
Good to know. I opted not to change this here as it made the test a bit messier, and the parent class already does more rigorous equality testing (`FieldTypeTestCase#testEquals`).